content/documentation/pages/3-stream-developer-guides/3-programming-models/0-index.md

```diff
@@ -5,6 +5,216 @@ description: 'Programming models'
 summary: true
 ---
 
-# Advanced Stream Processing
+# Programming Model for Streaming Applications
 
-Discuss at a high level what is covered in the following guides.
+Before we dive into the specifics of Spring Cloud Stream and the programming model choices, it is essential to discuss the design premise of Spring Cloud Stream.
+
+## Core Design
```

Comment:
The `core design` and the `binder abstraction` can be moved from the `streams concepts` as we discuss the same there.

Reply:
I don't see the details captured entirely in the "what are streams" section, but I removed it from here as per the comment.